common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -105,6 +105,11 @@ protected void handleMessage(
           for (int[] ids: msg.reduceIds) {
             numBlockIds += ids.length;
           }
+          // For fetching continuous shuffle blocks in batch, we store both start and end reduce
+          // id in reduceIds, so the real number of ShuffleBlockBatchId is total length / 2.
```

Comment:
I'd like to make the code more explicit
```
if (msg.batchFetchEnabled) {
  numBlockIds = msg.mapIds.length;
} else {
  for (int[] ids: msg.reduceIds) {
    numBlockIds += ids.length;
  }
}
```

Reply:
Writing the code in this way we don't even need a comment.

Reply:
Copy, done in 407b1e0.